node/impl/client/client.go

```diff
@@ -1011,13 +1022,28 @@ func (a *API) outputCAR(ctx context.Context, ds format.DAGService, bs bstore.Blo
 		cs := cid.NewSet()
 
 		for _, dagSpec := range dags {
+			dagSpec := dagSpec
+
 			if err := utils.TraverseDag(
 				ctx,
 				ds,
 				root,
 				dagSpec.selector,
+				func(node format.Node) error {
+					if dagSpec.exportAll {
+						lk.Lock()
```

Comment:
What is exportAll and how does it relate to retrieval and remote blockstores? It might benefit from a comment in code too.

Reply:
This is a fix for the merkle-tree export mode - previously we'd not export some nodes with some selectors - e.g. when doing unixfs hamt traversal.